src/MagicUrl.php

```diff
@@ -64,7 +64,7 @@ public function generate($endpoint)
     {
         return [
             'user'         => $this->user->ID,
-            'private'      => wp_hash_password($this->signature($endpoint)),
+            'private'      => wp_hash($this->signature($endpoint)),
```

Comment:
The timestamp/expiration should be part of the signature. 

Otherwise, it's possible to alter the transient expiration 

WordPress only "deletes" expired transients on fetching them.

Reply:
Don't rely on transient expiration for security. Enforce that in the code.

Reply:
Good call, I've updated the signature to include the `expires_at` timestamp 👍

Reply:
Remember to also update the check. 

There is some code duplication there in the validation part.

Reply:
> Remember to also update the check.

The expiration is part of the signature now, so it would be checked when comparing the hashes – did you mean something else?

> There is some code duplication there in the validation part.

Which part are you referring to?

Reply:
https://github.com/aaemnnosttv/wp-cli-login-command/blob/26d861eebd63bb34e95150ccf5bed8f993bba1fa/plugin/wp-cli-login-server.php#L314

The verification builds the signature without time.
So it's never valid, unless I'm missing something.

Reply:
Nvm, apparently I can't read. It's there.